src/RelayHooksTypes.ts

```diff
@@ -27,10 +27,27 @@ export type MutationConfig<T extends MutationParameters> = Partial<
     onCompleted?(response: T['response']): void;
 };
 
+export type MutationConfigWithoutVariables<T extends MutationParameters> = Omit<
+    MutationConfig<T>,
+    'variables'
+>;
+
+export type MutationConfigWithVariables<T extends MutationParameters> = MutationConfig<T> & {
+    variables: T['variables'];
+};
+
 export type Mutate<T extends MutationParameters> = (
     config?: Partial<MutationConfig<T>>,
 ) => Promise<T['response']>;
 
+export type MutateWithVariables<T extends MutationParameters> = (
+    config?: Partial<MutationConfig<T>> & { variables: T['variables'] },
```

Comment:
`config: Partial<MutationConfig<T>> & { variables: T['variables'] },`

Here it would be necessary to make the Config mandatory so as to set the variables